addons/mail/static/src/new/core_ui/message.xml

```diff
@@ -2,163 +2,165 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.message" owl="1">
-        <div class="o-mail-message position-relative py-1"
-            t-att-class="{
-                'o-highlighted-from-mention': message.isHighlightedFromMention,
-                'o-highlighted bg-view shadow-lg': props.highlighted,
-                'o-self-author': message.isSelfAuthored,
-                'o-selected': props.messageToReplyTo?.isSelected(props.thread, props.message),
-                'o-squashed pt-1': props.squashed,
-                'mt-1': !props.squashed and props.thread,
-                'px-3': !props.isInChatWindow,
-                'px-1': props.isInChatWindow,
-                'opacity-50': props.messageToReplyTo?.isNotSelected(props.thread, props.message),
-            }"
-            role="group"
-            t-att-aria-label="messageTypeText"
-            t-on-click="onClick"
-            t-on-mouseenter="onMouseenter"
-            t-on-mouseleave="onMouseleave"
-            t-ref="root"
-        >
-            <MessageInReplyTo t-if="message.parentMessage" alignedRight="isAlignedRight" message="message" onClick="props.onParentMessageClick"/>
-            <div class="o-mail-message-core position-relative d-flex flex-shrink-0">
-                <div class="o-mail-message-sidebar d-flex flex-shrink-0" t-att-class="{ 'justify-content-end': isAlignedRight }">
-                    <div t-if="!props.squashed" class="o-mail-avatar-container position-relative bg-view" t-on-click="ev => this.onClickAuthor(ev)">
-                        <img class="o-mail-message-author-avatar w-100 h-100 rounded-circle o_object_fit_cover"
-                             t-att-title="authorText"
-                             t-att-src="message.authorAvatarUrl"
-                             t-att-class="{ 'o_redirect cursor-pointer': hasAuthorClickable }" />
-                        <ImStatus t-if="message.author" persona="message.author" className="imStatusClassName"/>
+        <ActionSwiper onRightSwipe = "env.isSmall and isInInbox ? { action: () => this.messageService.setDone(this.message), bgColor: 'bg-success', icon: 'fa-check-circle' } : undefined">
```

Comment:
make sure condition on `isSmall` is "is device mobile" and not "small screen", otherwise it will not work as expected on tablets when changing orientation

Reply:
Done (use `isMobileOS`)